Source/EasyNetQ/JsonSerializer.cs

```diff
@@ -36,22 +44,31 @@ public byte[] MessageToBytes(Type messageType, object message)
             using (var jsonWriter = new Newtonsoft.Json.JsonTextWriter(streamWriter))
             {
                 jsonWriter.Formatting = jsonSerializer.Formatting;
+                jsonWriter.ArrayPool = JsonSerializerArrayPool<char>.Instance;
                 jsonSerializer.Serialize(jsonWriter, message, messageType);
             }
 
             return memoryStream.ToArray();
         }
 
         /// <inheritdoc />
-        public object BytesToMessage(Type messageType, byte[] bytes)
+        public object BytesToMessage(Type messageType, ReadOnlyMemory<byte> bytes)
         {
             Preconditions.CheckNotNull(messageType, "messageType");
-            Preconditions.CheckNotNull(bytes, "bytes");
 
-            using var memoryStream = new MemoryStream(bytes, false);
+            using var memoryStream = new MemoryStream(bytes.ToArray(), false); // TODO Do not copy here
```

Comment:
See this https://github.com/dotnet/runtime/issues/27156 and this https://github.com/dotnet/corefx/blob/6b9440735abc9c4e369376bca097766527971a7a/src/Common/src/System/IO/ReadOnlyMemoryStream.cs

Copy-paste approach will be not so bad for now :)

Reply:
Done